kratos/utilities/parallel_utilities.h

```diff
@@ -124,6 +125,61 @@ class BlockPartition
         return global_reducer.GetValue();
     }
 
+    /** @brief loop with thread local storage (TLS). f called on every entry in rData
+     * @param TThreadLocalStorage template parameter specifying the thread local storage
+     * @param f - must be a function accepting as input TContainerType::value_type& and the thread local storage
+     * Note: TLS is passed by value to make construction on the fly possible
+     */
+    template <class TThreadLocalStorage, class TFunction>
+    inline void for_each(const TThreadLocalStorage rThreadLocalStoragePrototype, TFunction &&f)
```

Comment:
not a const reference? i think that this way we make one more copy than what is strictly needed

Reply:
answer is [here](https://github.com/KratosMultiphysics/Kratos/pull/7261#discussion_r458071004)

Reply:
ok, still it implies an extra copy...maybe could we move it?

Reply:
so using `const TThreadLocalStorage &&rThreadLocalStoragePrototype`?

Reply:
> so using `const TThreadLocalStorage &&rThreadLocalStoragePrototype`?

ok this doesn't work :D

Reply:
Passing by ref works too!